### PR TITLE
[Auth] Update documentation to fill gaps

### PR DESCRIPTION
**Pull request by samhorlbeck**

### CODE REVIEW

File: packages-exp/auth-exp/src/core/auth/initialize.ts [hunk 1]
```diff
@@ -24,7 +24,31 @@ import { _fail } from '../util/assert';
...
+ *
+ * This function allows more control over the Auth instance than
+ * {@link getAuth}. `getAuth` will use platform-specific defaults to supply
+ * the {@link Dependencies}. In general, `getAuth` is the easiest way to
+ * initialize Auth and will work in most cases. Use `initializeAuth` if you
+ * need control over which persistence layer is used, or to minimize bundle
+ * size if you're not using either `signInWithPopup` or `signInWithRedirect`.
+ *
+ * For example, if your app only uses anonymous accounts and you only want
+ * accounts saved for the current session, you would initialize Auth thus:
```
Comment (egilmorez) [inline packages-exp/auth-exp/src/core/auth/initialize.ts:41]: Suggest just commanding and replacing "thus" with words describing the difference (thus is a little over formal for Google).

Like, "If your app uses only anonymous . . . , initialize it with <correct words>:"

File: packages-exp/auth-exp/src/core/credentials/phone.ts [hunk 2]
```diff
@@ -121,7 +121,7 @@ export class PhoneAuthCredential extends AuthCredential {
     return obj;
   }
 
-  /** {@inheritdoc AuthCredential.fromJSON} */
+  /** Generates a Phone credential based on a plain object or a JSON string. */
```
Comment (egilmorez) [inline packages-exp/auth-exp/src/core/credentials/phone.ts:124]: Literal?  Backtick if so.

File: packages-exp/auth-exp/src/core/providers/saml.ts [hunk 3]
```diff
@@ -46,6 +46,22 @@ export class SAMLAuthProvider extends FederatedAuthProvider {
     super(providerId);
   }
 
+  /**
+   * Generates an {@link AuthCredential} from a {@link UserCredential} after a
+   * successful SAML flow ompletes.
```
Comment (egilmorez) [inline packages-exp/auth-exp/src/core/providers/saml.ts:51]: Typo.

**Review by yuchenshi — APPROVED** [review]

File: packages-exp/auth-exp/src/model/public_types.ts [hunk 4]
```diff
@@ -1173,10 +1190,44 @@ export interface EmulatorConfig {
...
+ * The modular SDK enables tree shaking by allowing explicit declarations of
+ * dependencies. For example, a web app does not need to include code that
+ * enables Cordova redirect sign in. That functionality is therefore split into
+ * {@link browserPopupRedirectResolver} and
+ * {@link cordovaPopupRedirectResolver}. The dependencies object is how Auth is
+ * configured to reduce bundle sizes.
+ *
+ * There are two ways to initialize an auth instance: {@link getAuth} and
+ * {@link initializeAuth}. `getAuth` initializes everything using
+ * platform-specific configurations, while `initializeAuth` takes a Dependencies
```
Comment (egilmorez) [inline packages-exp/auth-exp/src/model/public_types.ts:1206]: Literal?

File: packages-exp/auth-exp/src/model/public_types.ts [hunk 5]
```diff
@@ -1173,10 +1190,44 @@ export interface EmulatorConfig {
...
+ * object directly, giving you more control over what is used.
+ *
  * @public
  */
 export interface Dependencies {
+  /**
+   * Which {@link Persistence} to use. If this is an array, the first
+   * Persistence that the device supports is used. The SDK will search for an
```
Comment (egilmorez) [inline packages-exp/auth-exp/src/model/public_types.ts:1214]: Literal `Persistence` here and below?  

Lots of "will" here too :)

**Review by egilmorez — COMMENTED** [review]
Some nits for you Sam, thanks!

File: packages-exp/auth-exp/src/core/credentials/phone.ts (hunk 2 above)
Comment (samhorlbeck) [reply packages-exp/auth-exp/src/core/credentials/phone.ts:124]: No literal here

File: packages-exp/auth-exp/src/core/auth/initialize.ts [hunk 6]
```diff
@@ -24,7 +24,31 @@ import { _fail } from '../util/assert';
...
 
-/** @public */
+/**
+ * Initializes an Auth instance with fine-grained control over
+ * {@link Dependencies}.
+ *
+ * @remarks
+ *
+ * This function allows more control over the Auth instance than
+ * {@link getAuth}. `getAuth` will use platform-specific defaults to supply
```
[the remark this answers is not in the document]
Comment (samhorlbeck) [reply packages-exp/auth-exp/src/core/auth/initialize.ts]: Done

File: packages-exp/auth-exp/src/core/auth/initialize.ts (hunk 1 above)
Comment (samhorlbeck) [reply packages-exp/auth-exp/src/core/auth/initialize.ts]: Done

File: packages-exp/auth-exp/src/core/providers/saml.ts (hunk 3 above)
Comment (samhorlbeck) [reply packages-exp/auth-exp/src/core/providers/saml.ts]: Done

File: packages-exp/auth-exp/src/model/public_types.ts (hunk 4 above)
Comment (samhorlbeck) [reply packages-exp/auth-exp/src/model/public_types.ts]: Done

File: packages-exp/auth-exp/src/model/public_types.ts (hunk 5 above)
Comment (samhorlbeck) [reply packages-exp/auth-exp/src/model/public_types.ts]: Done and done

File: packages-exp/auth-exp/src/core/credentials/phone.ts (hunk 2 above)
Comment (egilmorez) [reply packages-exp/auth-exp/src/core/credentials/phone.ts:124]: Should we lowercase "phone" in that case?

**Review by egilmorez — APPROVED** [review]
LGTM, thanks!

**Review by Feiyang1 — APPROVED** [review]

**Pull request merged**